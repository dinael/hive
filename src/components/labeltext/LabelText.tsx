import React, {FC} from 'react'
import styles from './LabelText.module.scss'

export type LabelTextProps = {
  text: string
  required?: boolean
  isRequiredText: string
  labelHidden?: boolean
  disabled?: boolean
  className?: string
}

export const LabelText: FC<LabelTextProps> = ({
  text,
  required = false,
  isRequiredText,
  labelHidden = false,
  disabled,
  className,
}: LabelTextProps) => {
  return (
    <span className={`
      ${styles['label-text-component']}
      ${styles[disabled ? 'is-disabled' : '']}
      ${styles[required ? 'is-required' : '']}
      ${labelHidden ? 'sr-only' : ''}
      ${className ? className : ''}
      `}>
      {text + ':'}
      {required ?
        <em
          className={styles['required-text']}
          title={isRequiredText? `${isRequiredText}` : 'is required'}>
          *
        </em>
        : null}
    </span>
  )
}

export default LabelText
