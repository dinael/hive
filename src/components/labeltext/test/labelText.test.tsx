import { describe, test, expect, vi } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'
import { LabelText, LabelTextProps } from '../LabelText'

describe('LabelText', ()=> {
  const defaultProps: LabelTextProps = {
    text: 'text-test',
    required: false,
    isRequiredText: 'isRequiredText-test',
    labelHidden: false,
    disabled: false,
    className: 'className-test',
  }

  test('renders snapshot', () => {
    const { container } = render(<LabelText {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

})
