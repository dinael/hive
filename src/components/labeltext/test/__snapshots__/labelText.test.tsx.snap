// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LabelText > renders snapshot 1`] = `
<div>
  <span
    class="
      label-text-component
      
      
      
      className-test
      "
  >
    text-test:
  </span>
</div>
`;
